cliquet/initialization.py

```diff
@@ -338,11 +338,14 @@ def initialize(config, version=None, project_name=None, default_settings=None):
         in application settings.
     :param dict default_settings: Override cliquet default settings values.
     """
-    if default_settings:
-        config.add_settings(default_settings)
-
     settings = config.get_settings()
 
+    if default_settings:
+        for key, value in default_settings.items():
+            if not settings.get(key):
```

Comment:
It is what @ametaireau was suggesting this morning, but since we are doing:

 - cliquet.DEFAULT_SETTINGS
 - read the user config.ini
 - kinto.DEFAULT_SETTINGS (updates keys from cliquet.DEFAULT_SETTINGS)

If the user is defining something that is inside the cliquet.DEFAULT_SETTINGS then kinto.DEFAULT_SETTINGS might force an unwanted settings.

Let say, cliquet defines: ``'cliquet.cache_backend': 'cliquet.cache.redis',`` and kinto propose ``'cliquet.cache_backend': 'cliquet.cache.memory',`` then it will be impossible to use redis with kinto because if the user configure redis it will be overrided to memory by kinto.